torch/_inductor/utils.py

```diff
@@ -479,15 +489,56 @@ def get_kernel_metadata(node_schedule, wrapper):
             key = node.meta["from_node"][0][0]
             from_node_dict[key].append(node.name)
     metadata = (
-        f"{wrapper.comment} Source Nodes: [{', '.join(sorted(from_node_dict.keys()))}], "
-        f"Original ATen: [{', '.join(sorted(original_aten_dict.keys()))}]"
+        f"{wrapper.comment} Source Nodes: [{', '.join(from_node_dict.keys())}], "
+        f"Original ATen: [{', '.join(original_aten_dict.keys())}]"
     )
+
     # trace back to original node here
-    detailed_metadata = []
+    detailed_metadata = [f"{wrapper.comment} Source node to ATen node mapping:"]
     for original_node, nodes in sorted(from_node_dict.items()):
         detailed_metadata.append(
-            f"{wrapper.comment} {original_node} => {', '.join(sorted(nodes))}"
+            f"{wrapper.comment}   {original_node} => {', '.join(sorted(nodes))}"
+        )
+
+    # print the aot_autograd graph fragment
+    detailed_metadata.append(f"{wrapper.comment} Graph fragment:")
+    for node in inductor_nodes:
+        node_s = node.format_node()
+        # Make the node string representation easier to read
+        # before: %abs_3 : [num_users=1] = call_function[target=torch.ops.aten.abs.default](args = (%tangents_1,), kwargs = {})
+        # after: %abs_3 = torch.ops.aten.abs.default(args = (%tangents_1,), kwargs = {})
```

Comment:
It'd be nice to use the Python code formatter but I'm not sure if you can conveniently only do a few nodes without having to shove them in an fx.Graph or something

Reply:
yeah, I hacked it after looking at how much work it would be to implement it properly and with the context that this PR should not be in the critical path, but can take another look if you prefer, lmk

Reply:
made it a bit less hacky